Firmware/sound.cpp

```diff
@@ -62,36 +62,38 @@ switch(eSoundMode)
 Sound_SaveMode();
 }
 
-//if critical is true then silend and once mode is ignored
-void Sound_MakeCustom(uint16_t ms,uint16_t tone_,bool critical){
+// Waits for the duration and uses the most appropriate delay type
+void Sound_Delay(uint16_t ms)
+{
+    if (ms < (SOUND_KEEPALIVE_MIN_MS))
```

Comment:
Not sure if doing something like this is really needed. You could just use delay_keep_alive(noUIupdates) regardless of duration. As long as no sounds are generated, delay_keep_alive() is just fine at all times, with the added bonus of managed heaters and inactivity messages on uart so that the host doesn't time out.

Reply:
I was not sure how much delay the update routine adds (which might also vary), so I thought it would make sense to keep the length of short beeps more accurate. But if this is not required, this could be simplified indeed :+1: